src/emulator/modules/SceGxm/SceGxm.cpp

```diff
@@ -1649,11 +1664,12 @@ EXPORT(int, sceGxmTextureInitTiled) {
     return unimplemented("sceGxmTextureInitTiled");
 }
 
-EXPORT(int, sceGxmTextureSetData, emu::SceGxmTexture *texture, Ptr<const void> data) {
-    assert(texture != nullptr);
-    assert(data);
-
-    texture->data = data;
+EXPORT(int, sceGxmTextureSetData, SceGxmTexture *texture, Ptr<const void> data) {
+    if (texture == nullptr){
+        return error("sceGxmTextureSetData", SCE_GXM_ERROR_INVALID_POINTER);
+    }
+    
+    texture->data_addr = (unsigned int)data.address() >> 2;
```

Comment:
This cast shouldn't really be necessary, as I think `address()` is a 32-bit int.